62.CAD/fragment_shader.hlsl

```diff
@@ -11,10 +12,84 @@ void beginInvocationInterlockEXT();
 void endInvocationInterlockEXT();
 #endif
 
+//#define NBL_DRAW_ARC_LENGTH
+
 // TODO[Lucas]: have a function for quadratic equation solving
 // Write a general one, and maybe another one that uses precomputed values, and move these to somewhere nice in our builtin hlsl shaders if we don't have one
 // See: https://github.com/erich666/GraphicsGems/blob/master/gems/Roots3And4.c
 
+template<typename float_t>
+struct DefaultClipper
+{
+    using float2_t = vector<float_t, 2>;
+    
+    inline float2_t operator()(const float t)
+    {
+        return clamp(t, 0.0, 1.0);
+    }
+};
+
+template<typename float_t>
+struct BezierLineStyleClipper

```

Comment:
Get `ArcLenCalculator` as a template param.
so this clipper is a general struct than can wotk with any curve type and not specialized to beziers

Reply:
done